sbt-app/src/sbt-test/dependency-management/sbt-plugin-publish/build.sbt

```diff
@@ -0,0 +1,116 @@
+import scala.util.matching.Regex
+
+lazy val repo = file("test-repo")
+lazy val resolver = Resolver.file("test-repo", repo)
+
+lazy val example = project.in(file("example"))
+  .enablePlugins(SbtPlugin)
+  .settings(
+    organization := "org.example",
+    addSbtPlugin("ch.epfl.scala" % "sbt-plugin-example-diamond" % "0.5.0"),
+    publishTo := Some(resolver),
+    checkPublishedArtifacts := checkPackagedArtifactsDef.value,
+    checkPublish := checkPublishDef.value
+  )
+
+lazy val testProject = project.in(file("test-project"))
+  .settings(
+    addSbtPlugin("org.example" % "example" % "0.1.0-SNAPSHOT"),
+    externalResolvers += {
+      val base = (ThisBuild / baseDirectory).value
+      MavenRepository("test-repo", s"file://$base/test-repo")
+    },
+    // ivyPaths := IvyPaths(baseDirectory.value, Some((ThisBuild / baseDirectory).value / "ivy-cache")),
+    // csrCacheDirectory := baseDirectory.value / "coursier-cache",
+    checkUpdate := checkUpdateDef(
+      "example_2.12_1.0-0.1.0-SNAPSHOT.jar",
+      "sbt-plugin-example-diamond_2.12_1.0-0.5.0.jar",
+      "sbt-plugin-example-left_2.12_1.0-0.3.0.jar",
+      "sbt-plugin-example-right_2.12_1.0-0.3.0.jar",
+      "sbt-plugin-example-bottom_2.12_1.0-0.3.0.jar",
+    ).value
+  )
+
+lazy val checkPublishedArtifacts = taskKey[Unit]("check the published artifacts")
+lazy val checkPublish = taskKey[Unit]("check publish")
+lazy val checkUpdate = taskKey[Unit]("check update")
+
+def checkPackagedArtifactsDef: Def.Initialize[Task[Unit]] = Def.task {
+  val packagedArtifacts = Keys.packagedArtifacts.value
+  val name = artifact.value.name
```

Comment:
Should we write `"example"` here to make sure the artifact `name` does not already contain the suffix?